go/vt/tabletserver/splitquery/splitter_test.go

```diff
@@ -0,0 +1,318 @@
+		message += fmt.Sprintf("len is different: expected: %v vs got:%v\n", len(expected), len(got))
+		return
+	}
+	for i := range expected {
+		if expected[i].Sql != got[i].Sql {
+			message += fmt.Sprintf("expected[%v].Sql:\n%v\n!=\ngot[%v].Sql:\n%v\n",
+				i, expected[i].Sql, i, got[i].Sql)
+		}
+		if expected[i].RowCount != got[i].RowCount {
+			message += fmt.Sprintf("expected[%v].RowCount: %v != got[%v].RowCount: %v\n",
+				i, expected[i].RowCount, i, got[i].RowCount)
+		}
+		if !reflect.DeepEqual(expected[i].BindVariables, got[i].BindVariables) {
+			message += fmt.Sprintf("expected[%v].BindVariables:\n%v\n!=\ngot[%v].BindVariables:\n%v\n",
+				i, expected[i].BindVariables, i, got[i].BindVariables)
+		}
+	}
+	t.Errorf("%s", message)
+}
+
+func TestSplit(t *testing.T) {
+	splitParams, err := NewSplitParams("select * from test_table",
+		map[string]interface{}{}, []string{"id", "user_id"}, getSchema())
+	if err != nil {
+		t.Fatalf("SplitParams.Initialize() failed with: %v", err)
+	}
+	splitter := NewSplitter(splitParams,
+		&FakeSplitAlgorithm{
+			boundaries: []tuple{
+				{Int64Value(1), Int64Value(2)},
+				{Int64Value(1), Int64Value(3)},
+				{Int64Value(5), Int64Value(1)},
+			},
+		})
+	var queryParts []querytypes.QuerySplit
+	queryParts, err = splitter.Split()
+	if err != nil {
+		t.Errorf("Splitter.Split() failed with: %v", err)
+	}
+	expected := []querytypes.QuerySplit{
+		{
+			Sql: "select * from test_table where" +
+				" (id < :_splitquery_end_id) or" +
+				" ((id = :_splitquery_end_id) and (user_id < :_splitquery_end_user_id))",
+			BindVariables: map[string]interface{}{
+				"_splitquery_end_id":      int64(1),
+				"_splitquery_end_user_id": int64(2),
+			},
+		},
+		{
+			Sql: "select * from test_table where" +
+				" ((:_splitquery_start_id < id) or" +
+				" ((:_splitquery_start_id = id) and (:_splitquery_start_user_id <= user_id)))" +
+				" and" +
+				" ((id < :_splitquery_end_id) or" +
+				" ((id = :_splitquery_end_id) and (user_id < :_splitquery_end_user_id)))",
+			BindVariables: map[string]interface{}{
+				"_splitquery_start_id":      int64(1),
+				"_splitquery_start_user_id": int64(2),
+				"_splitquery_end_id":        int64(1),
+				"_splitquery_end_user_id":   int64(3),
+			},
+		},
+		{
+			Sql: "select * from test_table where" +
+				" ((:_splitquery_start_id < id) or" +
+				" ((:_splitquery_start_id = id) and (:_splitquery_start_user_id <= user_id)))" +
+				" and" +
+				" ((id < :_splitquery_end_id) or" +
+				" ((id = :_splitquery_end_id) and (user_id < :_splitquery_end_user_id)))",
+			BindVariables: map[string]interface{}{
+				"_splitquery_start_id":      int64(1),
+				"_splitquery_start_user_id": int64(3),
+				"_splitquery_end_id":        int64(5),
+				"_splitquery_end_user_id":   int64(1),
+			},
+		},
+		{
+			Sql: "select * from test_table where" +
+				" (:_splitquery_start_id < id) or" +
+				" ((:_splitquery_start_id = id) and (:_splitquery_start_user_id <= user_id))",
```

Comment:
The parentheses around the inner AND factors and the comparators are not needed due to operator precedence. Only the outermost parentheses are needed.